sql/mysqld.cc

```diff
@@ -7669,6 +7671,9 @@ static int init_server_components() {
   if (initialize_storage_engine(default_tmp_storage_engine, " temp",
                                 &global_system_variables.temp_table_plugin))
     unireg_abort(MYSQLD_ABORT_EXIT);
+  if (initialize_storage_engine(default_dd_storage_engine, " data dictionary",
```

Comment:
Unfortunately this will not work, and at this stage it is too early to tell the correct fix. The default DD engine is special in that it has to be loaded significantly earlier in the startup process and its load method is diferent (`initialize_storage_engine` calls `ha_resolve_by_name` - which needs `mysql.plugin` table already accessible).

For example, in my experimental branch I load the plugin in a different way: https://github.com/laurynas-biveinis/mysql-5.6/commit/5838003fce1b77e06c1ea9b3657c031bb35c10ba:
- in CMakeLists.txt, the plugin has to be marked as `MANDATORY`, which makes it statically-linked too
- then it has to be handled in `plugin_register_builtin_and_init_core_se` in `sql_plugin.cc`.

But at this time I still cannot tell whether the above is sufficient too. I think we simply must revisit this at a later point.

Reply:
I removed this initialization and changed the sys var to enum. I hit some issue updating this PR. So end up with closing this one and creating a new PR.